src/format/opamFile.ml

```diff
@@ -437,6 +437,34 @@ module Pinned_legacy = LineFile(struct
 
   end)
 
+
+(** Cached environment updates (<switch>/environment) *)
+
+module Environment = LineFile(struct
+
+    let internal = "environment"
+
+    type t = env_update list
+
+    let empty = []
+
+    let pp =
+      Pp.lines_set ~empty:[] ~add:OpamStd.List.cons ~fold:List.fold_right @@
+      Pp.identity ^+
+      Pp.pp ~name:"env_update_op"
+        env_update_op_of_string string_of_env_update_op ^+
```

Comment:
My  bad, the `of_string` takes an additional `~pos` argument. You can either wrap in `(fun ~pos:_ -> xxx)`, or use `Pp.of_pair "env_update_op" (env_update_op_of_string, string_of_env_update_op)` instead.

Other than that, all good it seems. Thanks!